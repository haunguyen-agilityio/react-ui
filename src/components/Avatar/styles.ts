// lib MUI
import { Avatar } from '@mui/material'
import { styled } from '@mui/system'

// theme
import theme from '@themes/index'

export const StyledAvatar = styled(Avatar)({
  backgroundColor: '#21d083',
  fontWeight: 600,
  color: theme.palette.common.white,
})

export const SmallAvatar = styled(Avatar)({
  backgroundColor: '#aaaaaa',
})
