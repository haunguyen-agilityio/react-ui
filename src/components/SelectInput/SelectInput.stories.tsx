// Libraries
import { useState } from 'react'
import { ComponentMeta, ComponentStory } from '@storybook/react'
import { Box, MenuItem, Typography } from '@mui/material'
import { ThemeProvider } from '@mui/material/styles'

// Themes
import theme from '../../themes'

// Components
import SelectInput, { ISelectOptionProps } from '.'

const mockOptions: ISelectOptionProps[] = [
  {
    label: 'Option 1',
    value: 'option1',
  },
  {
    label: 'Option 2',
    value: 'option2',
  },
  {
    label: 'Option 3',
    value: 'option3',
  },
]

const mockAssetOptions: ISelectOptionProps[] = [
  {
    label: 'All',
    value: 'all',
  },
  {
    label: 'Bitcoin',
    value: 'bitcoin',
    source: '/icons/coins/bitcoin.svg',
  },
  {
    label: 'Ethereum',
    value: 'ethereum',
    source: '/icons/coins/ethereum.svg',
  },
  {
    label: 'Ripple',
    value: 'ripple',
    source: '/icons/coins/ripple.svg',
  },
  {
    label: 'Bitcoin Cash',
    value: 'bitcoin-cash',
    source: '/icons/coins/bitcoin-cash.svg',
  },
  {
    label: 'Litecoin',
    value: 'litecoin',
    source: '/icons/coins/litecoin.svg',
  },
]

const renderAssetOptions = mockAssetOptions.map(({ label, value, source }: ISelectOptionProps) => (
  <MenuItem key={value} value={value}>
    {source && (
      <Box className='asset-icon'>
        {/* <Image width={30} height={30} alt={label} src={source} /> */}
      </Box>
    )}
    <Typography className='asset-name'>{label}</Typography>
  </MenuItem>
))

const renderOptions = mockOptions.map((option: ISelectOptionProps) => (
  <MenuItem key={option.value} value={option.value}>
    {option.label}
  </MenuItem>
))

export default {
  title: 'Components/SelectInput',
  component: SelectInput,
  decorators: [
    (Story) => (
      <ThemeProvider theme={theme}>
        <Story />
      </ThemeProvider>
    ),
  ],
} as ComponentMeta<typeof SelectInput>

export const Playground: ComponentStory<typeof SelectInput> = (args) => {
  const [value, setValue] = useState('')

  const handleOnChange = (value: string) => {
    setValue(value)
  }

  return (
    <SelectInput {...args} value={value} onChange={handleOnChange}>
      {args.selectVariant === 'asset' ? renderAssetOptions : renderOptions}
    </SelectInput>
  )
}

export const All = () => (
  <Box
    sx={{
      display: 'flex',
      flexDirection: 'column',
      rowGap: '20px',
    }}
  >
    <SelectInput name='select-input-default' label='Default' value='' onChange={() => {}}>
      <MenuItem value=''>Default Value</MenuItem>
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='select-input-error'
      label='Error'
      value=''
      onChange={() => {}}
      errorMessage='Required'
      error
    >
      <MenuItem value=''>Default Value</MenuItem>
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='select-input-disabled'
      label='Disabled'
      value=''
      onChange={() => {}}
      disabled
    >
      <MenuItem value=''>Default Value</MenuItem>
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='select-input-selected'
      label='Selected An Option'
      value='option1'
      onChange={() => {}}
    >
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='small-select-input-default'
      label='Default Small'
      value=''
      onChange={() => {}}
      arrowIcon='secondary'
      size='small'
    >
      <MenuItem value=''>Default Value</MenuItem>
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='small-select-input-error'
      label='Error Small'
      value=''
      onChange={() => {}}
      errorMessage='Required'
      error
      arrowIcon='secondary'
      size='small'
    >
      <MenuItem value=''>Default Value</MenuItem>
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='small-select-input-disabled'
      label='Disabled Small'
      value=''
      onChange={() => {}}
      disabled
      arrowIcon='secondary'
      size='small'
    >
      <MenuItem value=''>Default Value</MenuItem>
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='small-select-input-selected'
      label='Selected An Option Small'
      value='option1'
      onChange={() => {}}
      arrowIcon='secondary'
      size='small'
    >
      {renderOptions}
    </SelectInput>
    <SelectInput
      name='asset-select-input-selected'
      label='Selected An Option (Asset)'
      value='bitcoin'
      onChange={() => {}}
      arrowIcon='secondary'
      selectVariant='asset'
    >
      {renderAssetOptions}
    </SelectInput>
    <SelectInput
      name='small-asset-select-input-default'
      label='Default An Option Small (Asset)'
      value=''
      onChange={() => {}}
      arrowIcon='secondary'
      selectVariant='asset'
      size='small'
    >
      <MenuItem value=''>Default Value</MenuItem>
      {renderAssetOptions}
    </SelectInput>
    <SelectInput
      name='small-asset-select-input-selected'
      label='Selected An Option Small (Asset)'
      value='bitcoin'
      onChange={() => {}}
      arrowIcon='secondary'
      selectVariant='asset'
      size='small'
    >
      {renderAssetOptions}
    </SelectInput>
  </Box>
)

export const Default = () => (
  <SelectInput name='select-input-default' label='Default' value='' onChange={() => {}}>
    <MenuItem value=''>Default Value</MenuItem>
    {renderOptions}
  </SelectInput>
)

export const Error = () => (
  <SelectInput
    name='select-input-error'
    label='Error'
    value=''
    onChange={() => {}}
    errorMessage='Required'
    error
  >
    <MenuItem value=''>Default Value</MenuItem>
    {renderOptions}
  </SelectInput>
)

export const Disabled = () => (
  <SelectInput name='select-input-disabled' label='Disabled' value='' onChange={() => {}} disabled>
    <MenuItem value=''>Default Value</MenuItem>
    {renderOptions}
  </SelectInput>
)

export const SelectedAnOption = () => (
  <SelectInput
    name='select-input-selected'
    label='Selected An Option'
    value='option1'
    onChange={() => {}}
  >
    {renderOptions}
  </SelectInput>
)

export const DefaultSmall = () => (
  <SelectInput
    name='small-select-input-default'
    label='Default Small'
    value=''
    onChange={() => {}}
    arrowIcon='secondary'
    size='small'
  >
    <MenuItem value=''>Default Value</MenuItem>
    {renderOptions}
  </SelectInput>
)

export const ErrorSmall = () => (
  <SelectInput
    name='small-select-input-error'
    label='Error Small'
    value=''
    onChange={() => {}}
    errorMessage='Required'
    error
    arrowIcon='secondary'
    size='small'
  >
    <MenuItem value=''>Default Value</MenuItem>
    {renderOptions}
  </SelectInput>
)

export const DisabledSmall = () => (
  <SelectInput
    name='small-select-input-disabled'
    label='Disabled Small'
    value=''
    onChange={() => {}}
    disabled
    arrowIcon='secondary'
    size='small'
  >
    <MenuItem value=''>Default Value</MenuItem>
    {renderOptions}
  </SelectInput>
)

export const SelectedAnOptionSmall = () => (
  <SelectInput
    name='small-select-input-selected'
    label='Selected An Option Small'
    value='option1'
    onChange={() => {}}
    arrowIcon='secondary'
    size='small'
  >
    {renderOptions}
  </SelectInput>
)

export const SelectAssetOption = () => (
  <SelectInput
    name='asset-select-input-selected'
    label='Selected An Option (Asset)'
    value='bitcoin'
    onChange={() => {}}
    arrowIcon='secondary'
    selectVariant='asset'
  >
    {renderAssetOptions}
  </SelectInput>
)

export const DefaultAssetOptionSmall = () => (
  <SelectInput
    name='small-asset-select-input-default'
    label='Default An Option Small (Asset)'
    value=''
    onChange={() => {}}
    arrowIcon='secondary'
    selectVariant='asset'
    size='small'
  >
    <MenuItem value=''>Default Value</MenuItem>
    {renderAssetOptions}
  </SelectInput>
)

export const SelectAssetOptionSmall = () => (
  <SelectInput
    name='small-asset-select-input-selected'
    label='Selected An Option Small (Asset)'
    value='bitcoin'
    onChange={() => {}}
    arrowIcon='secondary'
    selectVariant='asset'
    size='small'
  >
    {renderAssetOptions}
  </SelectInput>
)
