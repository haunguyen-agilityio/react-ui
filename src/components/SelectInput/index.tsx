// Libraries
import React, { ReactNode } from 'react'
import { SelectChangeEvent, FormControl } from '@mui/material'
import { ArrowDropDown, KeyboardArrowDown } from '@mui/icons-material'

// Styles
import { StyledFormHelperText, StyledFormLabel, StyledSelect } from './styles'

export interface ISelectOptionProps {
  label: string
  value: string
  source?: string
}

export interface ISelectInputProps {
  value: string
  name: string
  label?: string
  error?: boolean
  labelId?: string
  fullWidth?: boolean
  defaultValue?: string
  disabled?: boolean
  errorMessage?: string
  size?: 'medium' | 'small'
  selectVariant?: 'normal' | 'asset'
  arrowIcon?: 'primary' | 'secondary'
  children: ReactNode
  onChange: (value: string) => void
}

const SelectInput = ({
  value = '',
  label = '',
  name = 'select-input-name',
  error = false,
  labelId = 'select-input-labelid',
  fullWidth = true,
  defaultValue = '',
  disabled = false,
  errorMessage = '',
  size = 'medium',
  selectVariant = 'normal',
  arrowIcon = 'primary',
  children = <></>,
  onChange = () => {},
}: ISelectInputProps) => {
  const handleChange = (event: SelectChangeEvent<unknown>) => {
    onChange(event.target.value as string)
  }

  return (
    <FormControl fullWidth={fullWidth} disabled={disabled} error={error}>
      {label && (
        <StyledFormLabel id={labelId} error={error}>
          {label}
        </StyledFormLabel>
      )}
      <StyledSelect
        value={value}
        labelId={labelId}
        defaultValue={defaultValue}
        displayEmpty
        onChange={handleChange}
        disabled={disabled}
        name={name}
        size={size}
        error={error}
        IconComponent={arrowIcon === 'primary' ? ArrowDropDown : KeyboardArrowDown}
        selectVariant={selectVariant}
      >
        {children}
      </StyledSelect>
      {error && <StyledFormHelperText error={error}>{errorMessage}</StyledFormHelperText>}
    </FormControl>
  )
}

export default SelectInput
