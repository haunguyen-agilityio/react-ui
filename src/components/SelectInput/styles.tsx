// Libraries
import { styled } from '@mui/material/styles'
import { FormHelperText, FormLabel, Select, SelectProps } from '@mui/material'

// Themes
import theme from '../../themes'

interface IStyledSelect extends SelectProps {
  selectVariant: 'normal' | 'asset'
}

const StyledSelect = styled(
  (props: IStyledSelect) => (
    <Select
      {...props}
      MenuProps={{
        sx: {
          '& .MuiList-root': {
            padding: 0,
          },
          '& .MuiMenuItem-root': {
            borderBottom: `1px solid ${theme.palette.grey[300]}`,
            columnGap: theme.spacing(1.5),
            paddingTop: theme.spacing(1.2),
            paddingBottom: theme.spacing(1.2),
            '> .asset-icon': {
              display: 'flex',
              alignItems: 'center',
            },
            '&, > .asset-name': {
              fontSize: theme.typography.h4.fontSize,
            },
          },
        },
      }}
    />
  ),
  {
    shouldForwardProp: (prop) => prop !== 'selectVariant',
  },
)(({ theme, size, value, error, selectVariant }) => ({
  verticalAlign: 'middle',
  fontSize: theme.typography.h4.fontSize,
  borderRadius: 0,
  color: error
    ? theme.palette.error.main
    : size === 'small' || value !== ''
    ? theme.palette.common.black
    : theme.palette.grey[400],
  '& .MuiSvgIcon-root': {
    color: theme.palette.common.black,
  },
  '& fieldset': {
    borderColor: theme.palette.grey[400],
  },
  '&.Mui-error .MuiOutlinedInput-notchedOutline': {
    borderColor: theme.palette.error.main,
  },
  '&.Mui-focused .MuiOutlinedInput-notchedOutline': {
    borderColor: error ? theme.palette.error.main : theme.palette.grey[400],
  },
  '&.MuiOutlinedInput-root:hover fieldset': {
    borderColor: error ? theme.palette.error.main : theme.palette.grey[400],
  },
  '& .MuiSelect-select': {
    ...(selectVariant === 'asset' &&
      value !== '' && {
        display: 'flex',
        columnGap: theme.spacing(1),
        alignItems: 'center',
        paddingTop: theme.spacing(size === 'small' ? 0.75 : 1.3),
        paddingBottom: theme.spacing(size === 'small' ? 0.75 : 1.3),
        '> .asset-icon': {
          height: '20px',
          width: '20px',
        },
        '> .asset-name': {
          fontSize: theme.typography.h4.fontSize,
        },
      }),
  },
}))

const StyledFormHelperText = styled(FormHelperText)(({ theme }) => ({
  marginTop: theme.spacing(0.3),
  marginLeft: 0,
  fontSize: theme.typography.body1.fontSize,
}))

const StyledFormLabel = styled(FormLabel)(({ theme, error }) => ({
  color: theme.palette.common.black,
  fontSize: theme.typography.h5.fontSize,
  '&.Mui-focused': {
    color: error ? theme.palette.error.main : theme.palette.common.black,
  },
}))

export { StyledSelect, StyledFormLabel, StyledFormHelperText }
