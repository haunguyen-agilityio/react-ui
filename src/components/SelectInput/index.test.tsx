// Libraries
import renderer from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'
import { ThemeProvider } from '@mui/material/styles'
import { MenuItem, Typography, Box } from '@mui/material'

// Themes
import theme from '../../themes'

// Components
import SelectInput, { ISelectInputProps, ISelectOptionProps } from '.'

const mockOptions: ISelectOptionProps[] = [
  {
    label: 'Option 1',
    value: 'option1',
  },
  {
    label: 'Option 2',
    value: 'option2',
  },
  {
    label: 'Option 3',
    value: 'option3',
  },
]

const mockAssetOptions: ISelectOptionProps[] = [
  {
    label: 'All',
    value: 'all',
  },
  {
    label: 'Bitcoin',
    value: 'bitcoin',
    source: '/icons/coins/bitcoin.svg',
  },
  {
    label: 'Ethereum',
    value: 'ethereum',
    source: '/icons/coins/ethereum.svg',
  },
  {
    label: 'Ripple',
    value: 'ripple',
    source: '/icons/coins/ripple.svg',
  },
  {
    label: 'Bitcoin Cash',
    value: 'bitcoin-cash',
    source: '/icons/coins/bitcoin-cash.svg',
  },
  {
    label: 'Litecoin',
    value: 'litecoin',
    source: '/icons/coins/litecoin.svg',
  },
]

const renderOptions = mockOptions.map((option: ISelectOptionProps) => (
  <MenuItem key={option.value} value={option.value}>
    {option.label}
  </MenuItem>
))

const renderAssetOptions = mockAssetOptions.map(({ label, value, source }: ISelectOptionProps) => (
  <MenuItem key={value} value={value}>
    {source && (
      <Box className='asset-icon'>
        {/* <Image width={30} height={30} alt={label} src={source} /> */}
      </Box>
    )}
    <Typography className='asset-name'>{label}</Typography>
  </MenuItem>
))

describe('SelectInput', () => {
  const componentProps: ISelectInputProps = {
    name: 'test-select-input-name',
    labelId: 'test-select-input-labelid',
    onChange: jest.fn(),
    value: '',
    children: renderOptions,
  }
  const ThemedComponent = (props: ISelectInputProps) => (
    <ThemeProvider theme={theme}>
      <SelectInput {...props} />
    </ThemeProvider>
  )

  test('It should match DOM Snapshot when value is not empty', () => {
    const component = renderer
      .create(<ThemedComponent {...componentProps} value='option1' />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when value is empty', () => {
    const component = renderer.create(<ThemedComponent {...componentProps} />).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when error', () => {
    const component = renderer
      .create(<ThemedComponent {...componentProps} error errorMessage='Required' />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when not full width', () => {
    const component = renderer
      .create(<ThemedComponent {...componentProps} fullWidth={false} />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when select variant is asset select', () => {
    const component = renderer
      .create(
        <ThemedComponent {...componentProps} selectVariant='asset'>
          {renderAssetOptions}
        </ThemedComponent>,
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when small size', () => {
    const component = renderer.create(<ThemedComponent {...componentProps} size='small' />).toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It should match DOM Snapshot when use secondary icon', () => {
    const component = renderer
      .create(<ThemedComponent {...componentProps} arrowIcon='secondary' />)
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('It can be disabled', () => {
    const { getByDisplayValue } = render(
      <ThemedComponent {...componentProps} value='option1' disabled />,
    )
    expect(getByDisplayValue('option1')).toBeDisabled()
  })

  test('It has label', () => {
    const { getByText } = render(<ThemedComponent {...componentProps} label='Select label' />)
    expect(getByText('Select label')).toBeTruthy()
  })

  test('It triggers handleOnChange function when select an option', () => {
    const { getByDisplayValue } = render(<ThemedComponent {...componentProps} value='option1' />)
    fireEvent.change(getByDisplayValue('option1'), {
      target: { value: 'option2' },
    })
    expect(componentProps.onChange).toHaveBeenCalledTimes(1)
  })
})
