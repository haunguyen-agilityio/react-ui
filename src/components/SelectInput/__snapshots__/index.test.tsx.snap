// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectInput It should match DOM Snapshot when error 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-1mpz4w9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
  <p
    className="MuiFormHelperText-root Mui-error MuiFormHelperText-sizeMedium MuiFormHelperText-contained css-19jrxw0-MuiFormHelperText-root"
  >
    Required
  </p>
</div>
`;

exports[`SelectInput It should match DOM Snapshot when not full width 1`] = `
<div
  className="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dt91je-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`SelectInput It should match DOM Snapshot when select variant is asset select 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dt91je-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`SelectInput It should match DOM Snapshot when small size 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-21bsnc-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`SelectInput It should match DOM Snapshot when use secondary icon 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dt91je-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="KeyboardArrowDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`SelectInput It should match DOM Snapshot when value is empty 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1dt91je-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="notranslate"
      >
        ​
      </span>
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value=""
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`SelectInput It should match DOM Snapshot when value is not empty 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-21bsnc-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    onClick={[Function]}
  >
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      aria-labelledby="test-select-input-labelid mui-component-select-test-select-input-name"
      className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
      id="mui-component-select-test-select-input-name"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Option 1
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
      disabled={false}
      name="test-select-input-name"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value="option1"
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
      data-testid="ArrowDropDownIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m7 10 5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-hdw1oc"
      >
        <span
          className="notranslate"
        >
          ​
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
